molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/ExcelServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.molgenis.oneclickimporter.service.Impl;
+
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.molgenis.oneclickimporter.exceptions.EmptySheetException;
+import org.molgenis.oneclickimporter.exceptions.NoDataException;
+import org.molgenis.oneclickimporter.service.ExcelService;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+@Component
+public class ExcelServiceImpl implements ExcelService
+{
+	@Override
+	public List<Sheet> buildExcelSheetsFromFile(File file)
+			throws IOException, InvalidFormatException, NoDataException, EmptySheetException
+	{
+		Workbook workbook = WorkbookFactory.create(file);
+		int numberOfSheets = workbook.getNumberOfSheets();
+
+		List<Sheet> sheets = newArrayList();
+		for (int index = 0; index < numberOfSheets; index++)
```

Comment:
Can't

> A Workbook doesn't implement Iterable<Sheet> like a Sheet implements Iterable<Row>, so you can't use an "enhanced" for loop. But you can loop through the Sheet objects yourself, with a traditional for loop. Use getNumberOfSheets() to find the number of sheets, and getSheetAt to retrieve the individual Sheet objects.

@https://stackoverflow.com/questions/20340915/how-to-go-though-the-sheets-of-a-workbook-with-apache-poi